src/rgw/rgw_reshard.cc

```diff
@@ -325,42 +315,31 @@ int RGWBucketReshard::clear_index_shard_reshard_status(rgw::sal::RGWRadosStore*
   return 0;
 }
 
-static int create_new_bucket_instance(rgw::sal::RGWRadosStore *store,
+static int update_num_shards(rgw::sal::RGWRadosStore *store,
 				      int new_num_shards,
-				      const RGWBucketInfo& bucket_info,
-				      map<string, bufferlist>& attrs,
-				      RGWBucketInfo& new_bucket_info)
+				      RGWBucketInfo& bucket_info,
+				      map<string, bufferlist>& attrs)
 {
-  new_bucket_info = bucket_info;
-
-  store->getRados()->create_bucket_id(&new_bucket_info.bucket.bucket_id);
-
-  new_bucket_info.layout.current_index.layout.normal.num_shards = new_num_shards;
-  new_bucket_info.objv_tracker.clear();
-
-  new_bucket_info.new_bucket_instance_id.clear();
-  new_bucket_info.reshard_status = cls_rgw_reshard_status::NOT_RESHARDING;
-
-  int ret = store->svc()->bi->init_index(new_bucket_info);
-  if (ret < 0) {
-    cerr << "ERROR: failed to init new bucket indexes: " << cpp_strerror(-ret) << std::endl;
-    return ret;
+  if (!bucket_info.layout.target_index) {
+    bucket_info.layout.target_index.emplace();
   }
+  bucket_info.layout.target_index->layout.normal.num_shards = new_num_shards;
+
+  bucket_info.layout.resharding = rgw::BucketReshardState::NONE;
```

Comment:
it would be nice to set `resharding = InProgress` here, so we don't have to do a separate write for that in `BucketInfoReshardUpdate::start()`

Reply:
there seems to be a problem in BucketInfoReshardUpdate(), in that something is still making cls call to set_resharding_status(). yet to fix.

Reply:
done